### PR TITLE
refactor: refacto a11y check to prepare JSON export

### PR DESCRIPTION
**Pull request by soykje**
**TASK**: #2034 

### Description, Motivation and Context
In order to prepare data for Spark metrics dashboard, we want to refacto the actual a11y script.

### Types of changes
- [x] 🛠️ Tool
- [x] 🧪 Test
- [x] 🧠 Refactor

### Screenshots - Animations
image

image

### CODE REVIEW

**Comment by soykje** [comment]
:eyes: https://spark-ui-uhip-9tctv4ooo-sparkteams-projects.vercel.app/a11y-report.json

File: .github/workflows/quality.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 name: quality
+env:
+  FORCE_COLOR: 3
```
Comment (soykje) [inline .github/workflows/quality.yml:3]: This is mandatory, for colors to work in CI logs.

File: bin/check-a11y.js [hunk 2]
```diff
@@ -23,102 +23,181 @@ const pkgFilter = hasFilter
...
             'landmark-one-main': { enabled: false },
             region: { enabled: false },
             'color-contrast': { enabled: false },
           },
           runOnly: ['best-practice', 'wcag2a', 'wcag2aa', 'wcag21a', 'wcag21aa'],
-          reporter: 'no-passes',
+          reporter: 'v2',
         })
-        .exclude('.docblock-argstable')
+        .exclude('*:not(.docs-story)')
```
Comment (soykje) [inline bin/check-a11y.js:132]: Working with stories was not relevant anymore, and was causing some issues. We prefer now to parse the complete documentation page and define which DOM nodes we want to check.

File: bin/check-a11y.js [hunk 3]
```diff
@@ -23,102 +23,181 @@ const pkgFilter = hasFilter
...
+        ...rest,
+        [testedComponent]: {
+          timestamp,
+          ...(duplicate ? { url: [...duplicate.url, url] } : { url: [url] }),
+          ...(duplicate
+            ? { incomplete: mergeResultArrays(incomplete, duplicate.incomplete) }
+            : { incomplete: mergeResultArrays(incomplete) }),
+          ...(duplicate
+            ? { violations: mergeResultArrays(violations, duplicate.violations) }
+            : { violations: mergeResultArrays(violations) }),
```
Comment (andresin87) [inline bin/check-a11y.js:150]:
```suggestion
          timestamp,
          url: duplicate ? [...duplicate.url, url] : [url],
          incomplete: duplicate ? mergeResultArrays(incomplete, duplicate.incomplete) : mergeResultArrays(incomplete),
          violations: duplicate ? mergeResultArrays(violations, duplicate.violations) : mergeResultArrays(violations),
```

is not the same?

**Review by andresin87 — APPROVED** [review]

**Review by acd02 — APPROVED** [review]

File: bin/check-a11y.js (hunk 3 above)
Comment (soykje) [reply bin/check-a11y.js:150]: Yes it is the same, but I find this less readable.

**Pull request merged**